subsys/sdfw_services/ssf_client_notif.c

```diff
@@ -175,6 +192,10 @@ int ssf_client_notif_deregister(struct ssf_client_notif_listener *listener)
 		return -SSF_EINVAL;
 	}
 
+	if (ssf_client_notif_initialized == false) {
+		return -SSF_EPERM;
+	}
+
```

Comment:
I don't get it, we free the semaphore when deregistering?

Reply:
The semaphore is there to protect the `listeners` array and synchronize access to it from multiple threads.